### ISSUE TITLE
Task handles cancel on drop but are not `#[must_use]` — a fire-and-forget spawn dies silently

### ISSUE THREAD

**Issue by lexoliu** [lexoliu/executor-core#8]
`AnyLocalExecutorTask` and `AnyExecutorTask` cancel their future when dropped (the `Task` docs say so at `src/lib.rs`: "Dropping a task cancels its execution"), but neither struct is `#[must_use]`. A fire-and-forget spawn therefore compiles clean and dies on the spot:

```rust
spawn_local(async move { tick_animation().await });   // cancelled before it runs
```

Downstream this is WaterUI's documented "a background task never runs / an animation loop dies instantly" symptom; `waterui::task::spawn_local` re-exports `executor_core::spawn_local`, so the fix belongs here.

Add `#[must_use]` with a message naming `.detach()` to both erased task types. The message is the whole fix: awaiting, storing, or detaching the handle all satisfy the lint, and only the silent drop warns.
